app/liquid_tags/forem_tag.rb

```diff
@@ -0,0 +1,52 @@
+module ForemTag
+  REGISTRY_REGEXP = %r{#{URL.url}/\w+(/[\w-]+)?(/[\w-]+)?}
+  USER_ORG_REGEXP = %r{#{URL.url}/(?<name>[\w-]+)/?$}
+  POST_PODCAST_REGEXP = %r{#{URL.url}/(?<podcast>[\w-]+)/[\w-]+/?}
+  COMBINED_REGEXP = [USER_ORG_REGEXP, POST_PODCAST_REGEXP].freeze
+
+  def self.new(tag_name, input, parse_context)
+    link = process_input(input)
+    klass = determine_klass(link)
+    raise StandardError, "Invalid #{Settings::Community.community_name} URL." unless klass
+
+    klass.__send__(:new, tag_name, link, parse_context)
```

Comment:
Note: When I used
```ruby
klass.new(tag_name, link, parse_context)
```
I got a `private method new` error. So I decided to use this `__send__` approach, which is also [used here](https://github.com/forem/forem/blob/0e900fd4dd07f31796251864b265eacb7cfc2364/app/liquid_tags/unified_embed/tag.rb#L37) in the `tag.rb` class.

Reply:
Ooh, it makes a lot of sense to just route to it from inside `.new`! I love this!